SCAutolib/controller.py

```diff
@@ -476,20 +650,27 @@ def cleanup(self):
         opensc_module.restore()
 
     @staticmethod
-    def _validate_configuration(conf: dict, params: {} = None) -> dict:
+    def _validate_configuration(conf: dict, params: dict = None) -> dict:
         """
-        Validate schema of the configuration file. If some value is not present
-        in the config file, this value would be looked in the CLI parameters
+        Validates the schema of the provided configuration dictionary against
+        predefined schemas for CAs, users, and cards.
+        It also accounts for CLI parameters that might override or supplement
+        configuration file values.
 
-        :param conf: Configuration to be parsed (e.g. data loaded from
-            JSON file)
+        :param conf: The configuration data, typically loaded from a JSON file,
+                     to be validated.
         :type conf: dict
-        :param params: CLI arguments
-        :type params: dict
-        :return: dictionary with parsed values from conf and params attributes.
-            All values are retyped to specified type.
+        :param params: A dictionary of parameters (e.g., from CLI arguments)
+                       that might provide missing values or override existing
+                       ones in the ``conf`` dictionary.
+        :type params: dict, optional
+        :return: A dictionary containing the validated and potentially retyped
+                 configuration values.
```

Comment:
the `retyped` here sounds awkward. I would use `adjusted` or `overridden`:
```suggestion
        :return: A dictionary containing the validated and potentially adjusted
                 configuration values.
```
what do you think?